src/database/operations.rs

```diff
@@ -929,3 +959,78 @@ pub(crate) async fn upsert_repository(
     })
     .await
 }
+
+pub(crate) async fn get_merge_queue_pull_requests(
+    executor: impl PgExecutor<'_>,
+    repo: &GithubRepoName,
+    tree_priority: Option<i32>,
+) -> anyhow::Result<Vec<PullRequestModel>> {
+    measure_db_query("get_merge_queue_pull_requests", || async {
+        let records = sqlx::query_as!(
+            PullRequestModel,
+            r#"
+            SELECT
+                pr.id,
+                pr.repository as "repository: GithubRepoName",
+                pr.number as "number!: i64",
+                pr.title,
+                pr.author,
+                pr.assignees as "assignees: Assignees",
+                (
+                    pr.approved_by,
+                    pr.approved_sha
+                ) AS "approval_status!: ApprovalStatus",
+                pr.status as "pr_status: PullRequestStatus",
+                pr.priority,
+                pr.rollup as "rollup: RollupMode",
+                pr.delegated_permission as "delegated_permission: DelegatedPermission",
+                pr.base_branch,
+                pr.mergeable_state as "mergeable_state: MergeableState",
+                pr.created_at as "created_at: DateTime<Utc>",
+                try_build AS "try_build: BuildModel",
+                auto_build AS "auto_build: BuildModel"
+            FROM pull_request as pr
+            LEFT JOIN build AS try_build ON pr.build_id = try_build.id
+            LEFT JOIN build AS auto_build ON pr.auto_build_id = auto_build.id
+            WHERE pr.repository = $1
+              AND pr.status = 'open'
+              AND pr.approved_by IS NOT NULL
+              AND pr.mergeable_state = 'mergeable'
+              -- Tree closure check (if tree_priority is set)
+              AND ($2::int IS NULL OR pr.priority >= $2)
+            ORDER BY
```

Comment:
We should have a consistent queue order on the website and in the actual merge queue, and writing this logic in SQL is a bit complicated. I would thus implement the queue ordering in Rust, and in this query download all *potentially eligible* PRs (so those that are approved) without presorting them.